### PR TITLE
docs: update the table of supported hardware

### PR DESCRIPTION
**Pull request by do9xe**

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -41,8 +41,14 @@ Developer Documentation
    dev/configmode
    dev/wan
 
-Supported Devices
------------------
+Supported Devices & Architectures
+---------------------------------
+
+ar71xx-generic
+--------------
```
Comment (NeoRaider) [inline docs/index.rst]: Please use a lower heading level than the one in line 44/45...

File: docs/index.rst [hunk 2]
```diff
@@ -79,18 +96,51 @@ Supported Devices
...
+  - WNDR3700 (v1)
+
+mpc85xx-generic
+---------------
+
+* TP-Link
+  - TL-WDR4900 (v1)
+
+ramips-rt305x
+-------------
```
Comment (NeoRaider) [inline docs/index.rst]: `ramips-rt305x` is BROKEN so far as it doesn't support AP+Adhoc, it shouldn't be mentioned in the list of officially supported models.

Comment (do9xe) [inline docs/index.rst]: if it is broken, then why list it? We do not have the Archer C7 eiter.

Comment (NeoRaider) [inline docs/index.rst]: I just told you not to list it...

Comment (do9xe) [inline docs/index.rst]: Oh, I misread it as "mention that it is broken" ;) Ill change that.

File: docs/index.rst [hunk 3]
```diff
@@ -52,11 +58,20 @@ Supported Devices
 * D-Link
 
   - DIR-825 (B1)
+  - DIR-615 (C1)
+
+* GL-Inet
```
Comment (NeoRaider) [inline docs/index.rst]: Please test your changes before opening pull requests...

You need to add an empty line before starting a new sublist. This mistake was made all over the file...

Comment (NeoRaider) [inline docs/index.rst]: Well, I'll fix it up as I overlooked it before merging...

**Pull request merged**